### PR TITLE
Add a system plugin and a file system free space check.

### PR DESCRIPTION
**Pull request by fcami**
The system plugin is meant for operating-system level checks, like
checking for free space in file systems, amount of RAM and swap, etc.

The file system free space check is meant to avoid unfortunate ENOSPC
situations for DS and IPA.

### CODE REVIEW

**Comment by fcami** [comment]
The check now generates an entry whether there is an error to report or not.

File: src/ipahealthcheck/system/filesystemspace.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    def check(self):
+        for store in self._pathchecks:
+            percent = self.get_fs_free_space_percentage(store)
+            if percent < self.min_free_percent:
+                yield Result(
+                    self, constants.ERROR,
+                    msg='%s: %s %s%% < %s%%' % (
+                        store, 'free space percentage under threshold:',
+                        percent, self.min_free_percent
```
Comment (rcritten) [inline src/ipahealthcheck/system/filesystemspace.py]: Add in the variables used to generate the data into msg as keywords, so store=store, percent=percent, min_free=self.min_free_percent.

File: src/ipahealthcheck/system/filesystemspace.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    )
+                )
+            free_space = self.get_fs_free_space(store)
+            threshold = self._pathchecks[store]
+            if free_space < threshold:
+                yield Result(
+                     self, constants.ERROR,
+                     msg='%s: %s %s MiB < %s MiB' % (
+                         store, 'free space under threshold:',
+                         free_space, threshold
```
Comment (rcritten) [inline src/ipahealthcheck/system/filesystemspace.py]: Add in the variables used to generate the data into msg as keywords, so store=store, free_space=free_space, threshold=threshold

File: src/ipahealthcheck/system/filesystemspace.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                     )
+                )
+            else:
+                yield Result(
+                     self, constants.SUCCESS,
+                     msg='%s: %s %s MiB >= %s MiB' % (
```
Comment (rcritten) [inline src/ipahealthcheck/system/filesystemspace.py]: Other success don't return anything but passing but it doesn't hurt to pass additional data. You should include the variables used to calculate the outcome.

File: tests/test_system_filesystemspace.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+from base import BaseTest
+from unittest.mock import Mock
+from util import capture_results
+from collections import namedtuple
+
+from ipahealthcheck.core import config, constants
+from ipahealthcheck.system.plugin import registry
+from ipahealthcheck.system.filesystemspace import FileSystemSpaceCheck
+
+
```
Comment (rcritten) [inline tests/test_system_filesystemspace.py]: Can you add a success test case as well?

File: src/ipahealthcheck/system/filesystemspace.py (hunk 2 above)
Comment (fcami) [reply src/ipahealthcheck/system/filesystemspace.py]: Solved, I think.

File: src/ipahealthcheck/system/filesystemspace.py (hunk 3 above)
Comment (fcami) [reply src/ipahealthcheck/system/filesystemspace.py]: Solved, I think.

File: tests/test_system_filesystemspace.py (hunk 4 above)
Comment (fcami) [reply tests/test_system_filesystemspace.py]: Yes, doin' so

**Comment by rcritten** [comment]
logging is imported but not used.

**Comment by fcami** [comment]
re logging not used => removed.

File: tests/test_system_filesystemspace.py (hunk 4 above)
Comment (fcami) [reply tests/test_system_filesystemspace.py]: Done. There are three tests now.

**Pull request merged**